wfe2/wfe.go

```diff
@@ -2129,6 +2198,19 @@ func (wfe *WebFrontEndImpl) NewOrder(
 
 	logEvent.DNSNames = names
 
+	// TODO(#5545): Spending and Refunding can be async until these rate limits
+	// are authoritative. This saves us from adding latency to each request.
+	var ratelimitTxns []ratelimits.Transaction
+	var newOrderSuccessful bool
+	go func() {
+		ratelimitTxns = wfe.checkNewOrderLimits(ctx, acct.ID, names)
+	}()
+	defer func() {
+		if !newOrderSuccessful && ratelimitTxns != nil {
+			go wfe.refundNewOrderLimits(ctx, ratelimitTxns)
+		}
+	}()
```

Comment:
This could definitely lead to a data race, however, it matches what we're doing in NewAccount at the moment.